src/dock-api.js

```diff
@@ -99,10 +104,11 @@ export default class DockAPI {
       rpc = Object.assign(rpc, PoaRpcDefs);
     }
 
-    const isWebsocket = this.address && this.address.indexOf('http') === -1;
+    const addressStr = addressArray[0];
+    const isWebsocket = addressStr && addressStr.indexOf('http') === -1;
     const provider = isWebsocket
-      ? new WsProvider(this.address)
-      : new HttpProvider(this.address);
+      ? new WsProvider(addressArray)
+      : new HttpProvider(addressArray);
```

Comment:
`HttpProvider` doesn't accept an array as per its docs. Have you checked that it works?

Reply:
just updated this, thought maybe we had a test for it (we should at some point) but doesnt look like it

Reply:
created https://github.com/docknetwork/sdk/issues/419 for testing it in the future